src/superqt/_vendor/dataframeeditor/dataframeeditor.py

```diff
@@ -0,0 +1,1302 @@
+    return str
+
+
+def bool_false_check(value):
+    """
+    Used to convert bool entrance to false.
+    Needed since any string in bool('') will return True.
+    """
+    if value.lower() in _bool_false:
+        value = ""
+    return value
+
+
+def global_max(col_vals, index):
+    """Returns the global maximum and minimum."""
+    col_vals_without_None = [x for x in col_vals if x is not None]
+    max_col, min_col = zip(*col_vals_without_None)
+    return max(max_col), min(min_col)
+
+
+class BaseDialog(QDialog):
+    def __init__(self, parent=None):
+        QDialog.__init__(self, parent)
+
+    def set_dynamic_width_and_height(
+        self, screen_geometry, width_ratio=0.5, height_ratio=0.5
+    ):
+        """
+        Update width and height using an updated screen geometry.
+        Use a ratio for the width and height of the dialog.
+        """
+        screen_width = int(screen_geometry.width() * width_ratio)
+        screen_height = int(screen_geometry.height() * height_ratio)
+        self.resize(screen_width, screen_height)
+
+        # Make the dialog window appear in the center of the screen
+        x = int(screen_geometry.center().x() - self.width() / 2)
+        y = int(screen_geometry.center().y() - self.height() / 2)
+        self.move(x, y)
+
+    def show(self):
+        super().show()
+        window = self.window()
+        windowHandle = window.windowHandle()
+        screen = windowHandle.screen()
+        geometry = screen.geometry()
+        self.set_dynamic_width_and_height(geometry)
+        screen.geometryChanged.connect(self.set_dynamic_width_and_height)
+
+
+class DataFrameEditor(BaseDialog):
+    """
+    Dialog for displaying and editing DataFrame and related objects.
+    Based on the gtabview project (ExtTableView).
+    For more information please see:
+    https://github.com/wavexx/gtabview/blob/master/gtabview/viewer.py
+    """
+
+    CONF_SECTION = "variable_explorer"
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        # Destroying the C++ object right after closing the dialog box,
+        # otherwise it may be garbage-collected in another QThread
+        # (e.g. the editor's analysis thread in Spyder), thus leading to
+        # a segmentation fault on UNIX or an application crash on Windows
+        self.setAttribute(Qt.WA_DeleteOnClose)
+        self.is_series = False
+        self.layout = None
+
+    def setup_and_check(self, data, title: str = ""):
+        """
+        Setup DataFrameEditor:
+        return False if data is not supported, True otherwise.
+        Supported types for data are DataFrame, Series and Index.
+        """
+        self._model = None
+
+        self.layout = QGridLayout()
+        self.layout.setSpacing(0)
+        self.layout.setContentsMargins(20, 20, 20, 0)
+        self.setLayout(self.layout)
+        title = trans_(title)
+
+        if isinstance(data, pd.Series):
+            self.is_series = True
```

Comment:
for consistency with the rest of the library (and Qt) this should be a method `self.isSeries()` instead of `self.is_series` ...